docs/integrators.md

```diff
@@ -400,9 +400,158 @@ The `reb_integrator_mercurius` structure contains the configuration and data str
 `unsigned int safe_mode`
 :   If this flag is set to 1 (the default), the integrator will recalculate heliocentric coordinates and synchronize after every timestep to avoid problems with outputs or particle modifications between timesteps. Setting this flag to 0 will result in a speedup, but care must be taken to synchronize and recalculate coordinates manually if needed.
 
+## TRACE
 
+TRACE is a hybrid time-reversible integrator, based on the algorithm described in [Hernandez & Dehnen 2023](https://ui.adsabs.harvard.edu/abs/2023MNRAS.522.4639H/abstract). 
+It uses WHFast for long term integrations but switches time-reversibly to BS or IAS15 for all close encounters. TRACE is appropriate for systems with a dominant central mass that will occasionally have close encounters. 
+The TRACE implementation is described in UPDATE LINK [Lu et al in prep](https://ui.adsabs.harvard.edu/abs/2023MNRAS.522.4639H/abstract)
 
 
+The following code enables TRACE and sets the critical radius to 4 Hill radii
+=== "C"
+    ```c
+    struct reb_simulation* r = reb_create_simulation();
+    r->integrator = REB_INTEGRATOR_TRACE;
+    r->ri_trace.r_crit_hill = 4;
+    ```
+
+=== "Python"
+    ```python
+    sim = rebound.Simulation()
+    sim.integrator = "trace"
+    sim.ri_trace.r_crit_hill = 4
+    ```
+
+The `reb_integrator_trace` structure contains the configuration and data structures used by the hybrid symplectic TRACE integrator.
+
+`double (*S) (const struct reb_simulation* const r, const unsigned int i, const unsigned int j)`
+:   This is a function pointer to the switching function for close encounters between non-central bodies.
+    If NULL (the default), the default switching function will be used.
+    The arguments `i` and `j` are the indices of the two particles considered.
+    The return value is a scalar.
+    If this function returns a negative value, it means a close encounter has been flagged.
+
+    - Default switching function
+
+        This is a similar (but slightly modified) switching function used in MERCURY and MERCURIUS.
+
+        ```c
+        double reb_integrator_trace_switch_default(const struct reb_simulation* const r, const unsigned int i, const unsigned int j);           
+        ```
+
+    The switching function can be manually set using this syntax:
+
+    === "C"
+        ```c
+        struct reb_simulation* r = reb_create_simulation();
+        r->ri_trace.S = reb_integrator_trace_switch_default;
+        ```
+
+    === "Python"
+        ```python
+        sim = rebound.Simulation()
+        sim.ri_trace.S = "default"
+        ```
+        
+`double (*S_peri) (const struct reb_simulation* const r, const unsigned int j)`
+:   This is a function pointer to the switching function for close encounters involving the central body.
+    If NULL (the default), the default switching function will be used.
+    The argument `j` is the index of the non-central particle considered.
+    The return value is a scalar.
```

Comment:
For both switching functions' return value, we seem to use 0/1 in the code, but the documentation says pos/neg. Need to make this consistent.